File: NETMouse - .NET release/Examples/PascalABC.NET/Matrices_9.pas
{$reference ABCNET.dll}
uses ABCNET.Utils;
uses ABCNET.Extensions;
uses System;

function Sort(self: array [,] of integer): array [,] of integer; extensionmethod;
type
  T = array [,] of integer;
  
begin
  if self = nil then
    raise new ArgumentNullException('self');

  result := T(self.Clone());
  var rowsCount := result.GetLength(0);
  var colsCount := result.GetLength(1);
  for var j := 0 to colsCount - 1 do
    for var k := 0 to rowsCount - 1 do
      for var i := 0 to rowsCount - k - 2 do
        if result[i, j] > result[i + 1, j] then
          BaseU.Swap(result[i, j], result[i + 1, j]);
end;

begin
  Matr.Rand(BaseU.ReadInteger('N:'), BaseU.ReadInteger('M:'), -17, 26).Print().Sort().Print();
  BaseU.ReadChar();
end.

File: NETMouse - .NET release/Examples/C#/Matrices_9.cs
using ABCNET.Extensions;
using ABCNET.Utils;
using System;

namespace TestProgram
{
    internal static class Program
    {
        public static int[,] Sort(this int[,] matrix)
        {
            if (matrix == null)
                throw new ArgumentNullException(nameof(matrix));

            int[,] result = (int[,])matrix.Clone();
            int rowsCount = result.GetLength(0);
            int colsCount = result.GetLength(1);
            for (int j = 0; j < colsCount; j++)
                for (int k = 0; k < rowsCount; k++)
                    for (int i = 0; i < rowsCount - k - 1; i++)
                        if (result[i, j] > result[i + 1, j])
                            Base.Swap(ref result[i, j], ref result[i + 1, j]);

            return result;
        }

        private static void Main(string[] args)
        {
            Matr.Rand(Base.ReadInteger("N:"), Base.ReadInteger("M:"), -17, 26).Print().Sort().Print();
        }
    }
}